tests/test_spec_file_parser.py

```diff
@@ -122,21 +123,14 @@ def test_replace_macro_that_is_tag_name(self) -> None:
         See issue https://github.com/bkircher/python-rpm-spec/issues/33.
 
         """
-        spec = Spec.from_string(
-            r"""
-%global myversion 1.2.3
-Version: %{myversion}
-        """
-        )
-        assert replace_macros(spec.version, spec) == "1.2.3"
 
         spec = Spec.from_string(
             r"""
 %global version 1.2.3
 Version: %{version}
         """
         )
-        assert replace_macros(spec.version, spec) == "1.2.3"
+        spec.version == "1.2.3"
```

Comment:
Hey there! Sorry can't really merge this as is.

There are a couple of things in this commit that are totally OK for me. However, you seem to have changed the way spec file parsing works fundamentally here: you now replace macros _while_ parsing.

Actually, I think this is not a good thing because sometimes _I want_ to have the parsed spec _including_ the plain macros. If macros get substituted during parsing, there is no way of accessing them afterwards. These two distinct steps are the reason for two functions `Spec.from_file(filename)` and `replace_macros(string, spec)`